application-diagram-api/src/main/java/com/xwiki/diagram/internal/AbstractDiagramRunnable.java

```diff
@@ -53,20 +60,25 @@ public void addToQueue(DiagramQueueEntry queueEntry)
     }
 
     /**
-     * Create an empty blocking queue.
-     */
-    public void initilizeQueue()
-    {
-        this.diagramsQueue = new LinkedBlockingQueue<DiagramQueueEntry>();
-    }
-
-    /**
-     * Get the entries to be processed by the thread.
+     * Process new diagram entry of queue.
      * 
-     * @return queue with the entries for the thread
+     * @return queueEntry the new diagram entry
      */
-    public BlockingQueue<DiagramQueueEntry> getDiagramsQueue()
+    public DiagramQueueEntry processDiagram()
```

Comment:
I know I suggested ``processDiagram``, but looking at the final code I find it a bit misleading, since you don't do any processing inside this method. WDYT about ``nextDiagramQueueEntry()`` or something like this, to make it more clear that it just retrieves the next diagram queue entry.